crates/zksync/core/src/vm/db.rs

```diff
@@ -179,6 +198,11 @@ where
     <DB as Database>::Error: Debug,
 {
     fn read_value(&mut self, key: &StorageKey) -> zksync_types::StorageValue {
+        let address = key.address().to_address();
+        let slot = h256_to_u256(*key.key());
+        if let Some(read_accesses) = self.read_accesses.lock().unwrap().as_mut() {
+            read_accesses.entry(address).or_insert_with(Vec::new).push(slot.to_alloy());
```

Comment:
We should use a `HashSet` (or `OrderedHashSet` if ordering is important) and push the `(address, slot)` directly.

Reply:
I though about it but in EVM there is the case that they want duplicated entries of slots